dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserAccountService.java

```diff
@@ -67,92 +67,115 @@ public class DefaultUserAccountService implements UserAccountService {
   private static final int MAX_LENGTH = 80;
   private static final int MAX_PHONE_NO_LENGTH = 30;
 
+  @Override
+  public void validateSelfRegUser(UserRegistrationParams userRegParams, String remoteAddress)
```

Comment:
I am also wondering what kind of registration other than a "self" registration there is? We should find a simple language for this, like "registration" is a user driven registration, "invite" a admin driven registration.